Python/ceval.c

```diff
@@ -2376,20 +2376,19 @@ void
 PyEval_SetProfileAllThreads(Py_tracefunc func, PyObject *arg)
 {
     PyThreadState *this_tstate = _PyThreadState_GET();
-    PyInterpreterState* interp = this_tstate->interp;
+    PyInterpreterState *interp = this_tstate->interp;
 
-    _PyRuntimeState *runtime = &_PyRuntime;
-    HEAD_LOCK(runtime);
+    INTERP_HEAD_LOCK(interp);
     PyThreadState* ts = PyInterpreterState_ThreadHead(interp);
-    HEAD_UNLOCK(runtime);
+    INTERP_HEAD_UNLOCK(interp);
```

Comment:
This is wrong, you cannot use per interp lock to traverse interpreters from runtime, it should stay to use global lock